### ISSUE TITLE
Install poetry dependencies without main package

### ISSUE THREAD

**Issue by miracle2k** [sdispater/poetry#516]
I do this in my Dockerfile:

```
RUN pip install poetry
COPY poetry.lock .
COPY pyproject.toml .
RUN poetry install

COPY appmodule ./appmodule
```

This is a nice technique to have Docker cache the dependencies as long as they don't change. This worked perfectly well until recently. The newest versions of poetry complain:

```
[ValueError]
No file/folder found for package apppackage

Exception trace:
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/poetry/console/commands/install.py in handle() at line 67
   builder = SdistBuilder(self.poetry, NullEnv(), NullIO())
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/poetry/masonry/builders/builder.py in __init__() at line 35
   includes=self._package.include,
 /Users/michael/.pyenv/versions/3.7.0/lib/python3.7/site-packages/poetry/masonry/utils/module.py in __init__() at line 51
   raise ValueError("No file/folder found for package {}".format(name))
```

It appears I can add

```
RUN mkdir photoworld_server && touch apppackage/__init__.py
```

to my Dockerfile to make it work again, but it seems like a hack and I think poetry should support this natively.

**Comment by jacebrowning** [comment]
I think this might be a duplicate of https://github.com/sdispater/poetry/issues/512.

**Comment by dmytrokyrychuk** [comment]
There are rumors that this has been fixed in 0.12.3. @miracle2k, can you confirm?

**Comment by jacebrowning** [comment]
It's fixed for me in 0.12.3!

**Comment by pawamoy** [comment]
Still got this issue with Poetry 0.12.10, had to copy my src folder.
